lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -89,3 +89,137 @@ class Meta:
             'startDateTime',
             'endDateTime',
         )
+
+
+class LtiAgsScoreSerializer(serializers.ModelSerializer):
```

Comment:
>I've decided to do a bunch of the validation/checking on the serializer itself, so there shouldn't ever be a record in the DB with a null score_given or score_maximum. All record manipulation (timestamp validation and deletion of existing records when no score is provided) happens as a result of the serializer.

The new implementation is more complex and only handle cases through the LTI-AGS API, and doesn't put any safeguards in the model (we'll have internal python APIs later).

> so there shouldn't ever be a record in the DB with a null score_given or score_maximum.

Adding this artificial requirement only makes it harder to implement the actual working endpoint. Please remove the validators from the serializer and allow `score_given` and `score_maximum` to store null values.

This needs to be a valid request:
```
{
  "timestamp": "2017-04-16T18:54:36.736+00:00",
  "activityProgress" : "Completed",
  "gradingProgress" : "PendingManual",
  "userId" : "5323497"
}
```
And it won't work in the current implementation.

Reply:
@giovannicimolin 
In a situation where a user has a score, the score is 'erased' using a record with a blank `scoreGiven` (thus saving `null` values to the database for the blanks), and then a request is made to the results endpoint, should the endpoint return 1 record with a blank score, or no records?

Reply:
The LTI spec mentions that the Results service can ignore any score entry that doesn't have a final grade:
https://www.imsglobal.org/spec/lti-ags/v2p0/#results-url-may-skip-empty-results

So this simplifies a lot our implementation, and don't need to return any result.